src/pages/messenger.tsx

```diff
@@ -0,0 +1,254 @@
+import { useState, useRef, useCallback, ChangeEvent, useEffect } from 'react';
+
+// components
+import ToggleSwitch from 'components/ToggleSwitch';
+import Header from 'components/Header';
+// import MessengerList from 'components/MessageList';
+
+// utils
+import useInput from 'hooks/useInput';
+import { useRecoilValue } from 'recoil';
+import { userGroup } from 'utils/atom';
+import { setDateFormat } from 'utils';
+import { ChatMessageType } from 'utils/type';
+
+// style
+import styled from 'styled-components';
+import { flexCenter } from 'styles/theme';
+import Send from 'assets/icon-send.png';
+import Save from 'assets/icon-save.png';
+import { theme } from 'styles/theme';
+
+// constant
+const JSON_FILENAME = 'chatting_info.json';
+
+const Messenger = () => {
+  const [curUserId, setCurUserId] = useState<number>(1);
+  const [otherUserId, setOtherUserId] = useState<number>(2);
+  const [messageStorage, setMessageStorage] = useState<ChatMessageType[]>([
+    {
+      userId: 1,
+      data: 'sample',
+      timestamp: Date(),
+    },
+  ]);
+
+  const input = useInput('');
+  const curUserGroup = useRecoilValue(userGroup);
+  const scrollRef = useRef<null | HTMLDivElement>(null);
+
+  const handleUserToggle = useCallback(
+    (e: ChangeEvent<HTMLInputElement>) => {
+      setOtherUserId(curUserId);
+      setCurUserId(otherUserId);
+    },
+    [curUserId, otherUserId]
+  );
+
+  const scrollToBottom = () => {
+    if (scrollRef.current) {
+      scrollRef.current.scrollTop = scrollRef.current.scrollHeight;
+    }
+  };
+
+  useEffect(() => {
+    scrollToBottom();
+  }, [messageStorage]);
+
+  const resetInput = (e: any) => {
```

Comment:
```suggestion
  const resetInput = (e: React.FormEvent<HTMLFormElement>) => {
```
와 같이 수정할 수 있을 것 같아요~
vscode에서 해당 속성에 마우스 호버를 하면 이벤트를 추천해주니 이를 활용해서 이벤트 타입을 선언하면 아주 편합니다!

Reply:
오 이벤트 타입 선언할 때 골치아팠는데 마우스 호버해서 타입을 가져와야겠네요 팁 감사합니당 ㅎㅎ 👍